### DISCUSSION TITLE
Code reuse non-OJS?

### DISCUSSION THREAD

**Discussion by DamonCharlesRoberts** [quarto-dev/quarto-cli#4367] [Q&A] [answered]
There might be a way to do this and I am not seeing it, but I was wondering if there is a way that I could do a [code-reuse the same way that you can with OJS objects](https://quarto.org/docs/interactive/ojs/code-reuse.html) but with Python objects?

For example:

`file_1.qmd`:
```
a = 1
```

`file_2.qmd`

```
import a from "./file_1.qmd"

a + 1
```

Or should I just go the standard route and define `a` in `file_1.py`?

**Comment by cscheid** [comment]
Are those code cells meant to be like so?

````
```{python}
# file_1.qmd
a = 1
```
````

````
```{python}
# file_2.qmd
import a from "./file_1.qmd"

a + 1
```
````

In this case, no, that's not supported and is unlikely going to be supported in the near-to-medium future. The fundamental reason is that we try to keep a pretty close relationship between .qmd and the .ipynb it can produce, and the cell in `file_2.qmd` isn't valid Python code.

**Reply by DamonCharlesRoberts** [reply]
Totally understandable! Thank you!

**Comment by baptiste** [comment, marked as answer]
For R chunks, a way to achieve this is to use code externalisation, i.e. the code chunks are in a separate R file, and all qmd files can refer to them. [Example gist](https://gist.github.com/baptiste/cdd86df690597d98947ce811e45b98fb)

It's also possible to generate the external R file automatically from a qmd document using `knitr::purl()`, so the process could be automated (`file2.qmd` referring to chunks from `file1.qmd` via an automatically generated R file, which could be done as a pre-render step, for instance. When doing this with several files, things could get a bit messy with chunk names (you'd have to keep a good convention to make them unique) and execution order.
